src/back/glsl.rs

```diff
@@ -743,7 +802,14 @@ fn write_expression<'a, 'b>(
             module.borrow_type(builder.locals[handle].ty),
         ),
         Expression::Load { pointer } => {
-            write_expression(&builder.expressions[pointer], module, builder)?
+            let (expr, ty) = write_expression(&builder.expressions[pointer], module, builder)?;
+
+            let ty = match *ty.borrow() {
+                TypeInner::Pointer { base, .. } => base,
```

Comment:
I think what we want is to not have pointers where they are implied. I.e. when accessing any global or local variables, doing pointer expression indirection is pointless